.github/workflows/dependabot-auto-approve.yml

```diff
@@ -0,0 +1,22 @@
+name: Dependabot auto-approve
+on: pull_request
+
+permissions:
+  pull-requests: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      - name: Approve a PR
+        if: ${{(contains(steps.metadata.outputs.dependency-names, 'eslint') || contains(steps.metadata.outputs.dependency-names, '@types')) && (steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor')}}
```

Comment:
This condition is very long. Is there a way to split it into multiple lines? I think YAML does support multi-line values, although I don't know the syntax.

Would something like this work? (Based on https://stackoverflow.com/q/3790454/69868)

```yaml
if: >
  ${{(
    contains(steps.metadata.outputs.dependency-names, 'eslint') ||   
    contains(steps.metadata.outputs.dependency-names, '@types')
  ) && (
    steps.metadata.outputs.update-type == 'version-update:semver-patch' ||   
    steps.metadata.outputs.update-type == 'version-update:semver-minor'
  )}}
```

It would be even nicer if we could provide the list of pre-approved dependencies as an array, so that we don't have to repeat the part `contains(steps.metadata.outputs.dependency-names`. This is nice to have though.

Maybe we could auto-approve Dependabot PRs for all `devDependencies` specified in `package.json`?

Anyhow, it's probably best to start with what you have and then improve this initial solution incrementally over time.

Reply:
Good suggestion with the multiline string. I had tried it but couldn't figure out the right syntax. 🙏 

I've also added a `matrix` so that we configure a list of dependencies once.

> Maybe we could auto-approve Dependabot PRs for all devDependencies specified in package.json?

I think this is too wide a change. For example with `electron` we still need manual review. So far I've only added the linter and ts utilities, which for minor/patch changes are safe to me.

Reply:
> I've also added a `matrix` so that we configure a list of dependencies once.

I don't see this change in the diff. Did you perhaps forget to push some commits?